test/functional/mongo_client_options.test.js

```diff
@@ -110,6 +110,28 @@ describe('MongoClient Options', function() {
     });
   });
 
+  it('NODE-2874: connectTimeoutMS=0 causes monitoring to time out', function(done) {
```

Comment:
nit: the jira ticket number will be there in the commit associated with adding this test, I think it clutters the test output a bit to include it here